workspaces/adventure-pack/goodies/typescript/Iterator.prototype.flatMap/index.ts

```diff
@@ -0,0 +1,31 @@
+import "../Iterator.prototype.toIterable";
+import { iteratorPrototype } from "../Iterator.prototype";
+
+declare global {
+  interface Iterator<T> {
+    flatMap<TOut>(
+      this: Iterator<T>,
+      callbackFn: (
+        element: T,
+        index: number,
+      ) => Iterator<TOut> | Iterable<TOut>,
+    ): Generator<TOut, void, undefined>;
+  }
+}
+
+iteratorPrototype.flatMap ??= function* <TIn, TOut>(
+  this: Iterator<TIn>,
+  callbackFn: (element: TIn, index: number) => Iterator<TOut> | Iterable<TOut>,
```

Comment:
Would be nice to have a test that uses the `index`, if we don't have one already.